api/lib/domain/models/Scorecard.js

```diff
@@ -26,6 +27,36 @@ class Scorecard {
     this.status = status;
   }
 
+  static buildFrom({ userId, userKEList, competence, competenceEvaluations }) {
+    const sortedKEGroupedByCompetence = _.groupBy(userKEList, 'competenceId');
+    const knowledgeElementsOfCompetence = sortedKEGroupedByCompetence[competence.id];
+    const totalEarnedPixByCompetence = _.sumBy(knowledgeElementsOfCompetence, 'earnedPix');
+
+    return new Scorecard({
+      id: `${userId}_${competence.id}`,
+      name: competence.name,
+      description: competence.description,
+      competenceId: competence.id,
+      index: competence.index,
+      area: competence.area,
+      earnedPix: totalEarnedPixByCompetence,
+      status: Scorecard._computeStatus({ knowledgeElements: knowledgeElementsOfCompetence, competenceId: competence.id, competenceEvaluations })
+    });
+  }
+
+  static _computeStatus({ knowledgeElements, competenceId, competenceEvaluations }) {
```

Comment:
C'est paradoxale d'avoir une méthode préfixée par `_` ce qui suggère qu'elle est privée, et de l'exporter en static. Si l'idée derrière c'était de montrer qu'elle n'utilise pas `this` je l'aurai simplement déclarée en fonction privée.

Reply:
Je vois qu'elle est testée unitairement dans des tests. Dans ce cas je l'aurai laissé en `static` mais j'aurai enlevé le `_` puisqu'elle n'est plus privée ;)

Reply:
oui le `_` laissé est un oubli